### PR TITLE
Iss951 - allow CICSTSManager SPI to expose create new terminal

### PR DESCRIPTION
**Pull request by hobbit1983**

### CODE REVIEW

File: galasa-managers-parent/buildSrc/src/main/groovy/galasa.java.gradle [hunk 1]
```diff
@@ -10,8 +10,7 @@ sourceCompatibility = 1.8
 targetCompatibility = 1.8
 
 java {
-    withJavadocJar()
```
Comment (Michael-Baylis) [inline galasa-managers-parent/buildSrc/src/main/groovy/galasa.java.gradle:13]: need javadoc and sources for maven

File: galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.vtp.manager/src/main/java/dev/galasa/vtp/internal/properties/VtpAPI.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import dev.galasa.vtp.manager.VtpManagerException;
+
+/**
+ * Should the VTP Manager use the VTP API to control VTP
+ * 
+ * @galasa.cps.property
+ * 
+ * @galasa.name vtp.API.enable
+ * 
+ * @galasa.description Should the VTP manager use the VTP API
```
Comment (Michael-Baylis) [inline galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.vtp.manager/src/main/java/dev/galasa/vtp/internal/properties/VtpAPI.java:17]: why is API capatalised?   we are leaving capitals for dynamic data

File: galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.vtp.manager/src/main/java/dev/galasa/vtp/manager/VtpManagerException.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ * Copyright contributors to the Galasa project
+ */
+package dev.galasa.vtp.manager;
+
+import dev.galasa.ManagerException;
+
+public class VtpManagerException extends ManagerException {
+
+	public VtpManagerException() {
+		// TODO Auto-generated constructor stub
```
Comment (Michael-Baylis) [inline galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.vtp.manager/src/main/java/dev/galasa/vtp/manager/VtpManagerException.java:11]: strip the todos please

File: galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.vtp.manager/src/main/java/dev/galasa/vtp/manager/internal/VtpManagerImpl.java [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+	private Path storedArtifactRoot;
+	
+	private VtpRecorderImpl recorder;
+
+	@Override
+	public void provisionGenerate() throws ManagerException, ResourceUnavailableException {
+		List<AnnotatedField> foundAnnotatedFields = findAnnotatedFields(CicstsManagerField.class);
+		for (AnnotatedField annotatedField : foundAnnotatedFields) {
+			Field field = annotatedField.getField();
+			if (field.getType() == ICicsRegion.class) {
```
Comment (Michael-Baylis) [inline galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.vtp.manager/src/main/java/dev/galasa/vtp/manager/internal/VtpManagerImpl.java:81]: no way.
The VTP Manager MUST not touch the cics manager annotation or fields.
If the VTP Manager requires a list of CICS regions,  then it should as for them via the SPI.

Comment (Michael-Baylis) [reply galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.vtp.manager/src/main/java/dev/galasa/vtp/manager/internal/VtpManagerImpl.java:81]: if you need an annotaion to indicate a link to a cics region,   then the vtp manager should have it's own annotation

**Review by Michael-Baylis — CHANGES_REQUESTED** [review]
Please make sure every file ends with a NL.
Minor changes, requested.
But one glaring bad practice accessing the cics manager annotation and types

File: galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.vtp.manager/src/main/java/dev/galasa/vtp/manager/VtpManagerException.java (hunk 3 above)
Comment (hobbit1983) [reply galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.vtp.manager/src/main/java/dev/galasa/vtp/manager/VtpManagerException.java:11]: done

File: galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.vtp.manager/src/main/java/dev/galasa/vtp/internal/properties/VtpAPI.java (hunk 2 above)
Comment (hobbit1983) [reply galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.vtp.manager/src/main/java/dev/galasa/vtp/internal/properties/VtpAPI.java:17]: good point - caught between an acronym and dynamic data

**Review by Michael-Baylis — APPROVED** [review]

**Pull request merged**